applications/bci_visualization/operators/reconstruction/convert_to_voxels_operator.py

```
@@ -0,0 +1,189 @@
+    out: NDArray[np.float32] = np.array([])  # bind outside loop
+    B: NDArray[np.float32] = np.array([])  # bind outside loop
+    while ctr < 100:
+        ctr += 1
+        inds = rng.choice(np.arange(len(ijk)), size=n, replace=False)
+        ins = ijk[np.array(inds), :]  # <- points
+        out = xyz[np.array(inds), :]  # <- mapped to
+        B = np.vstack([np.transpose(ins), np.ones(n, dtype=np.float32)])
+        if np.linalg.det(B) == 0:
+            continue
+    if np.linalg.det(B) == 0:
+        raise RuntimeError("Cannot compute affine, algorithm failed after 100 attempts")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)